### PR TITLE
update invisible wallet tutorial code

### PR DESCRIPTION
**Pull request by aaronTufts**

### CODE REVIEW

File: invisible-wallet/typescript/package.json [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
   "devDependencies": {
-    "typescript": "^5.0.4"
+    "typescript": "^5.2.2"
   },
   "dependencies": {
-    "@mysten/sui.js": "^0.41.0",
+    "@mysten/sui.js": "^0.45.1",
+    "@shinami/clients": "^0.6.0",
```
Comment (jasonxh) [inline invisible-wallet/typescript/package.json]: There's a newer version.

File: invisible-wallet/typescript/src/invisible_wallet_examples.ts [hunk 2]
```diff
@@ -1,169 +1,197 @@
...
+const walletClient = new WalletClient(ALL_SERVICES_TESTNET_ACCESS_KEY);
+const nodeClient = createSuiClient(ALL_SERVICES_TESTNET_ACCESS_KEY);
+
+// 5. Set up some variables we'll use in the examples below
+let sessionToken = "";
+let senderAddress = "";
+let senderSignature = null;
+
+// 6. Create the first wallet by generating a session token using its assocated secret
+sessionToken = await keyClient.createSession(WALLET_ONE_SECRET);
```
Comment (jasonxh) [inline invisible-wallet/typescript/src/invisible_wallet_examples.ts]: I suggest promoting `ShinamiWalletSigner` to abstract away the session management like this:
https://github.com/shinamicorp/shinami-typescript-sdk/tree/main/packages/clients#invisible-wallet

Also I suggest us thinking through our strategy for maintaining sample code. We have at least these 3 places:
- SDK readmes / examples
- Tutorials
- This repo

It'll become more and more difficult for us to keep them in sync, and confusing for users to see different versions.

File: invisible-wallet/typescript/package.json (hunk 1 above)
Comment (aaronTufts) [reply invisible-wallet/typescript/package.json]: Updated both `@mysten/sui.js" and "@shinami/clients" for both the invisible wallet and gas station. Will be reflected in my next push.

File: invisible-wallet/typescript/src/invisible_wallet_examples.ts (hunk 2 above)
Comment (aaronTufts) [reply invisible-wallet/typescript/src/invisible_wallet_examples.ts]: I've updated my code to use  ShinamiWalletSigner. I'll test it, update the tutorial copy and code to reflect this, and then push my updates.

Also, good point about maintenance. I have a story in the backlog of the dev docs epic to pick up soon to do more planning around the next phase of the docs and have added this to it.

File: invisible-wallet/typescript/src/invisible_wallet_examples.ts [hunk 3]
```diff
@@ -1,169 +1,170 @@
...
+  WALLET_ONE_ID,
+  walletClient,
+  WALLET_ONE_SECRET,
+  keyClient
+);
+// Returns the Sui address if it creates a wallet, otherwise undefined if the wallet exists
+let addressOrUndefined = await signer.tryCreate();
+
+// 7. If undefined, we can look up the  address with the `gatAddress` method
+const WALLET_ONE_SUI_ADDRESS = addressOrUndefined ? addressOrUndefined : await signer.getAddress();
```
Comment (jasonxh) [inline invisible-wallet/typescript/src/invisible_wallet_examples.ts]: You can directly use `signer.getAddress(true)` which saves you the trouble of calling `signer.tryCreate()`.

Comment (aaronTufts) [reply invisible-wallet/typescript/src/invisible_wallet_examples.ts]: updated

**Review by jasonxh — APPROVED** [review]
🚀🚀🚀

**Pull request merged**